sentry-core/src/main/java/io/sentry/core/EventCachedEventFireAndForgetIntegration.java

```diff
@@ -0,0 +1,125 @@
+package io.sentry.core;
+
+import static io.sentry.core.ILogger.logIfNotNull;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.concurrent.Executors;
+
+public final class EventCachedEventFireAndForgetIntegration implements Integration {
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  @Override
+  public void register(IHub hub, SentryOptions options) {
+    String cachedDir = options.getCacheDirPath();
+    if (cachedDir == null) {
+      logIfNotNull(
+          options.getLogger(), SentryLevel.WARNING, "No cache dir path is defined in options.");
+      return;
+    }
+
+    File outbox = new File(options.getCacheDirPath());
+    try {
+      Executors.callable(
+              () -> {
+                SendCachedFiles(hub, options.getLogger(), options.getSerializer(), outbox);
+              })
+          .call();
+    } catch (Exception e) {
+      logIfNotNull(
+          options.getLogger(),
+          SentryLevel.ERROR,
+          "Failed trying to send cached events at %s",
+          outbox);
+    }
+  }
+
+  private static void SendCachedFiles(
+      IHub hub, ILogger logger, ISerializer serializer, File directory) {
+    if (!directory.exists()) {
+      logIfNotNull(
+          logger,
+          SentryLevel.INFO,
+          "Directory '%s' doesn't exist. Nothing no cached events to send.",
+          directory.getAbsolutePath());
+      return;
+    }
+    if (!directory.isDirectory()) {
+      logIfNotNull(
+          logger,
+          SentryLevel.ERROR,
+          "Cache dir %s is not a directory.",
+          directory.getAbsolutePath());
+      return;
+    }
+
+    logIfNotNull(
+        logger,
+        SentryLevel.DEBUG,
+        "Processing %d items from cache dir %s",
+        directory.length(),
+        directory.getAbsolutePath());
+
+    for (File file : directory.listFiles()) {
+      // TODO: postfix should ve a const, shared with the caching code (still to be merged)
+      if (!file.getName().endsWith(".sentry-event")) {
+        logIfNotNull(
+            logger,
+            SentryLevel.DEBUG,
+            "File '%s' doesn't match extension expected.",
+            file.getName());
+        continue;
+      }
+
+      if (!directory.isFile()) {
+        logIfNotNull(logger, SentryLevel.DEBUG, "'%s' is not a file.", directory.getAbsolutePath());
+        return;
+      }
+
+      if (!file.canRead()) {
```

Comment:
if you want to delete a file, the way how to check is `file.getParentFile().canWrite()`

Reply:
Good point. We do need to read it too. But I guess if we can't delete it, maybe we ignore the file otherwise we would be sending this file on each app start.